### PR TITLE
[SITE] Refactoring Docs/ folder

### PR DESCRIPTION
**Pull request by naskio**
- Refactoring Docs/ folder in order to match the naming convention.

### CODE REVIEW

**Comment by m4rs-mt** [comment]
One of the recent contributions changed the existing syntax highlighting functionality. Therefore, this PR has some conflicts. @naskio Can you resolve the current conflicts and push and updated version?

File: Docs/04_Upgrade-Guides/04_v0.6.X-to-v0.7.X.md [hunk 1]
```diff
@@ -98,20 +109,22 @@ class ...
...
-    static void ...(GroupedIndex index)
-    {
-        // Common ILGPU way (still supported)
-        int gridIdx = index.GridIdx;
-        int groupIdx = index.GroupIdx;
+static void ...(GroupedIndex index)
+{
+// Common ILGPU way (still supported)
+int gridIdx = index.GridIdx;
+int groupIdx = index.GroupIdx;
```
Comment (NullandKale) [inline Docs/04_Upgrade-Guides/04_v0.6.X-to-v0.7.X.md:123]: The white spaces here are important to mimic the correct indentation for c# code.

Comment (NullandKale) [reply Docs/04_Upgrade-Guides/04_v0.6.X-to-v0.7.X.md:123]: This change should be reverted

**Review by NullandKale — CHANGES_REQUESTED** [review]
Everything other than the one change to white space for the c# like code formatting is good.

Not super important other than changing the one spelling mistake and renaming a few files.

I will approve this merge if the comment can be addressed

File: Docs/04_Upgrade-Guides/04_v0.6.X-to-v0.7.X.md (hunk 1 above)
Comment (naskio) [reply Docs/04_Upgrade-Guides/04_v0.6.X-to-v0.7.X.md:123]: ` ```c#` and ` ``` ` were missing here.

**Review by m4rs-mt — COMMENTED** [review]
Looks good to me know. However, seems to be the case that there are still some conflicts to address.

**Comment by naskio** [comment]
> Looks good to me know. However, seems to be the case that there are still some conflicts to address.

Can you please re-check now ?

**Review by m4rs-mt — APPROVED** [review]

**Pull request merged**